src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -60,7 +60,7 @@ internal static byte[] SslGetAlpnSelected(SafeSslHandle ssl)
             int len;
             SslGetAlpnSelected(ssl, out protocol, out len);
 
-            if (len == 0)
+            if (len < 1)
```

Comment:
len might be negative?

Reply:
who knows its coming from unmanaged code, I was merely adding extra safety. It might be worth me explaining the hole

Reply:
@Priya91, CryptoNative_SslGet0AlpnSelected should be initializing `len` always; any of our shims that are used with out vars should initialize the value to the default.  Otherwise the calling C# code using out might end up with garbage in the value yet C# will allow the value to be used.

Reply:
@Drawaes, I don't think this `len < 1` is correct or the right fix.  The way the native shim function is currently coded, if for example `HAVE_OPENSSL_ALPN` isn't defined or if `!API_EXISTS(SSL_get0_alpn_selected)`, the P/Invoke is a nop and `len` won't be modified, leaving it at whatever garbage was in `len` from the stack here, which means `len` could be anything, including a garbage value `>=1`.  The right fix would seem to be to ensure that `CryptoNative_SslGet0AlpnSelected` initializes `len` to 0 if it's not calling `SSL_get0_alpn_selected`, and then this call site can return to a check for `== 0`.

Reply:
@Drawaes Will you be making this change in this PR?